main.tf

```diff
@@ -126,8 +138,8 @@ data "aws_iam_policy_document" "assume_role_policy" {
   }
 }
 
-data "aws_iam_policy_document" "allows_s3" {
-  count = local.enabled ? 1 : 0
+data "aws_iam_policy_document" "s3_access_for_sftp_users" {
+  for_each = local.enabled ? local.user_names_map : {}
```

Comment:
Couldn't this use local.user_names instead? Why is local.user_names_map required?

Reply:
`for_each` requires a `set` or `map` and `user_names` is a `list`